icu4c/source/common/mlbe.h

```diff
@@ -60,31 +61,48 @@ class MlBreakEngine : public UMemory {
     void loadMLModel(UErrorCode &error);
 
     /**
-     * Initialize the element list from the input string.
+     * In the machine learning's model file, specify the name of the key and value to load the
+     * corresponding feature and its score.
+     *
+     * @param rb A ResouceBundle corresponding to the model file.
+     * @param keyName The kay name in the model file.
+     * @param valueName The value name in the model file.
+     * @param index The index of the fModel.
+     * @param error Information on any errors encountered.
+     */
+    void initKeyValue(UResourceBundle *rb, const char *keyName, const char *valueName,
+                      int32_t index, UErrorCode &error);
+
+    /**
+     * Initialize the index list from the input string.
      *
      * @param inString A input string to be segmented.
-     * @param elementList A list to store the first six characters.
+     * @param indexList A list to store the code unit index of the first six characters.
      * @param status Information on any errors encountered.
      * @return The number of code units of the first six characters in inString.
      */
-    int32_t initElementList(const UnicodeString &inString, UChar32* elementList,
-                            UErrorCode &status) const;
+    int32_t initIndexList(const UnicodeString &inString, int32_t *indexList,
+                          UErrorCode &status) const;
 
     /**
      * Evaluate whether the index is a potential breakpoint.
      *
-     * @param elementList A list including six elements for the breakpoint evaluation.
+     * @param inString A input string to be segmented.
+     * @param indexList A list including six code unit indexes of the characters for the breakpoint
+     * evaluation.
      * @param index The breakpoint index to be evaluated.
+     * @param codeUts The code unit boundary of the indexList.
      * @param numBreaks The accumulated number of breakpoints.
      * @param boundary A vector including the index of the breakpoint.
      * @param status Information on any errors encountered.
      */
-    void evaluateBreakpoint(UChar32* elementList, int32_t index, int32_t &numBreaks,
-                            UVector32 &boundary, UErrorCode &status) const;
+    void evaluateBreakpoint(const UnicodeString &inString, int32_t *indexList, int32_t index,
+                            int32_t codeUts, int32_t &numBreaks, UVector32 &boundary,
+                            UErrorCode &status) const;
 
     UnicodeSet fDigitOrOpenPunctuationOrAlphabetSet;
     UnicodeSet fClosePunctuationSet;
-    Hashtable fModel;
+    Hashtable fModel[13];
```

Comment:
please add a note about the 13. how about

Hashtable fModel[13]; // {UW1, UW2, ... UW6, BW1, ... BW3, TW1, TW2, ... TW4} 6+3+4= 13

Reply:
Done